lib/readline.js

```diff
@@ -580,27 +580,48 @@ Interface.prototype._wordLeft = function() {
 Interface.prototype._wordRight = function() {
   if (this.cursor < this.line.length) {
     var trailing = this.line.slice(this.cursor);
-    var match = trailing.match(/^(?:\s+|\W+|\w+)\s*/);
+    var match = trailing.match(/^(?:\s+|[^\w\s]+|\w+)\s*/);
     this._moveCursor(match[0].length);
   }
 };
 
+function charLengthLeft(str, i) {
+  if (i <= 0)
+    return 0;
+  if (i > 1 && str.codePointAt(i - 2) > 2 ** 16 ||
+      str.codePointAt(i - 1) > 2 ** 16) {
```

Comment:
I made a mistake when suggesting this: it should be `>=` instead of `>`. Otherwise the character `𐀀` would not be properly handled. The same applies for `charLengthAt`. Sorry about that.

Reply:
@BridgeAR should be fixed now

Reply:
I meant all `str.codePointAt(i - x) > 2 ** 16` should be `str.codePointAt(i - x) >= 2 ** 16`. Otherwise the code point which is exactly 2 ** 16 won't be detected properly.

Reply:
sorry miss read that